app/controllers/application_controller.rb

```diff
@@ -33,6 +33,12 @@ def current_provider
                           end
   end
 
+  # Returns the default role
+  def default_role
+    default_role_setting = SettingGetter.new(setting_name: 'DefaultRole', provider: current_provider).call
+    @default_role = Role.find_by(name: default_role_setting)
```

Comment:
And if we can't find the "default_role" we may have to default to the provider "User" role which is stable and not meant to be removed, right?

Reply:
> Besides, what is the use of the instance variabe @default_role?

For consistency as currrent_provider and current_user are returning an instance variable. I had the same question TBH. Does not need to return an instance variable.

Reply:
> 



> Things that we may need to check:
> 
> 1. The role name uniqueness per provider scope.
> 2. A user record cannot have a role from a different provider.
> 3. the default_role if not found we fallback to the current_provider "User" role.
> 4. "User" role cannot be deleted even when no record depends on it (I believe this is a WIP)
> 5. Each provider should have its essential roles created? "User, Guest & administrator"

1. Check. `populate_roles` migration in db/data
2. Check in current PR.
3. Should always be found since the default value of the DefaultRole SiteSetting is set to 'User' in `populate_default_role_setting` migration in db/data. Could add redundancy to ApplicationController#default_role.
4. Done in a previous PR. "Guest" "Administrator" and "User" can't be deleted.
5. Check. `populate_roles` migration in db/data

Reply:
I think I may want to synchronize with you on the concerns addressed above and their scope,
I believe that we may not have solved all of the issues,